salt/salt.spec

```diff
@@ -352,8 +352,8 @@ Requires:       iproute
 %endif
 
 %if %{with systemd}
-BuildRequires:  pkgconfig(systemd)
```

Comment:
@brejoc I am not sure about this change.

I think we can keep `pkgconfig(systemd)` for both all OS, including Fedora.

But I am not sure about what `systemd_ordering` is or why we this adds `systemd_requires`.

@EthanB11 can you explain?

Reply:
Ah, I see the removed lines were added here: https://github.com/juliogonzalez/salt-packaging/commit/7705b22d7d29986b9ba2daf0743caaa04f262493

Reply:
So it seems `BuildRequires:  systemd` is incorrect and we need to keep `BuildRequires:  pkgconfig(systemd)`.

Still I am not sure about the `%{?systemd_ordering}` to `%{?systemd_requires}` change.

Reply:
https://github.com/systemd/systemd/blob/master/src/core/macros.systemd.in#L31
```

%systemd_requires \
Requires(post): systemd \
Requires(preun): systemd \
Requires(postun): systemd \
%{nil}

%systemd_ordering \
OrderWithRequires(post): systemd \
OrderWithRequires(preun): systemd \
OrderWithRequires(postun): systemd \
%{nil}
```
https://rpm-packaging-guide.github.io/

> The OrderWithRequires tag for supplying additional ordering hints has been added. The tag follows Requires tag syntax, but does not generate actual dependencies. The ordering hints are treated as if they were Requires when calculating the transaction order, only if the involved packages are present in the same transaction.

Seems to me we don't need `systemd_ordering` but `systemd_requires` as @EthanB11 did?

Reply:
What line number is that? I just compared my salt.spec to the one from here https://build.opensuse.org/package/view_file/systemsmanagement:Uyuni:Master:CentOS8-Uyuni-Client-Tools/salt/salt.spec?expand=1 and I dont see any line change regarding your comment.

> Still I am not sure about the %{?systemd_ordering} to %{?systemd_requires} change.

All the changes I see are the if statements adding Fedora and another 3 lines requiring dnf like centos8 also has.

Reply:
I do see something I missed in that all the requirements for fedora may not be listed. I see a few lines where rhel 8 has requirements that Fedora does not have.

Reply:
I have no clue about that, but I'm sure @DimStar77 could help!

Reply:
That is basically reverting my previous change - which is really not nice.

pkgconfig(systemd) is provided by the package called systemd (the package shipping systemd.pc) - in openSUSE, we have two packages doing so (systemd and systemd-mini; the -mini only used in build- environments, but having a much cheaper dependency chain). Reverting that is, at least on openSUSE, not welcome at all.

as for ```%systemd_ordering``` vs ```%systemd_requires`: systemd_ordering means 'if the systems is installed together with systemd, instruct the package manager to first install systemd - but in fact, do not force systemd'; salt does not forcibly require systemd to be usable on a system, which makaes ```%systemd_requires``` the wrong choice (this is true for almost all services out there)

Reply:
Yeah, the first thing about `pkconfig(systemd)` I got it and I agree it should stay.

Not sure I understand the second one: How is it that salt does not forcibly require systemd for systems where systemd is available?

Reply:
@EthanB11

At https://build.opensuse.org/package/view_file/systemsmanagement:Uyuni:Master:CentOS8-Uyuni-Client-Tools/salt/salt.spec?expand=1, starting line 357:
```
%if %{with systemd}
BuildRequires:  pkgconfig(systemd)
%{?systemd_ordering}
%else
```
At https://build.opensuse.org/package/view_file/home:ethanb11:Fedora_Uyuni:Master:Fedora30-Uyuni-Client-Tools/salt/salt.spec?expand=1 starging at line 354:
```
%if %{with systemd}
BuildRequires:  systemd
%{?systemd_requires}
%else
```

But maybe it's because you branched the package **before** @DimStar77 change was merged. That explains the difference if you didn't changed such lines.

Reply:
salt is not forcibly requiring systemd (full stop) - systemd is the service manager (of course you know that) used by most distros. But salt does not actually care/need systemd.

systemd_requires thought adds an explicit dependency on systemd - which makes systemd not 'uninstallable on systems where you don't want to see it);

with systemd_ordering the statement is: if this is a systemd managed system, make sure systemd is install in the transaction to install packages prio to salt - so that systemd is present at the time our scriptlets run (the scriptlets are the only thing really needing systemd - more so than salt even would on its own)

Reply:
> Yeah, the first thing about `pkconfig(systemd)` I got it and I agree it should stay.
> 
> Not sure I understand the second one: How is it that salt does not forcibly require systemd for systems where systemd is available?

Anyway, rolling back this part of the changes.